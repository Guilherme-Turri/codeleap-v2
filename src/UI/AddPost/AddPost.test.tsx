import { componentRender } from '../../render/componentRender'
import { AddPost } from '.'
import { screen } from '@testing-library/react'
describe('<AddPost />', () => {
  it('should render Post when isClicked is true', () => {
    const { container } = componentRender(<AddPost isClicked={true}></AddPost>)
    const close = screen.getByTestId('Close')
    expect(close).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })

  it('should render AddPost when isClicked is false', () => {
    componentRender(<AddPost isClicked={false}></AddPost>)
    const addPost = screen.getByTestId('AddPost')
    expect(addPost).toBeInTheDocument();
  })
})