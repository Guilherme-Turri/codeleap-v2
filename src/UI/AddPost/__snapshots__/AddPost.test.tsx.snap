// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPost /> should render Post when isClicked is true 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  color: #00CBA9;
  width: 100%;
  height: 100%;
}

.c0 {
  color: #c4d299;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="c0"
    data-testid="AddPost"
  >
    <div
      class="c1"
      data-testid="Close"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
        >
          <path
            d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm2.71 11.29a1 1 0 0 1 0 1.42 1 1 0 0 1-1.42 0L12 13.41l-1.29 1.3a1 1 0 0 1-1.42 0 1 1 0 0 1 0-1.42l1.3-1.29-1.3-1.29a1 1 0 0 1 1.42-1.42l1.29 1.3 1.29-1.3a1 1 0 0 1 1.42 1.42L13.41 12z"
            data-name="close-circle"
          />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
